### PR TITLE
app-arch/cpio, app-arch/tar: install as `gcpio` and `gtar`

### PR DESCRIPTION
**Pull request by mgorny**
Do it like FreeBSD and partially like Fedora because:

1. It's ugly to do it for Prefix only (i.e. Prefix installs have `gtar`, normal don't).
2. If we do it, then `gtar` should be the actual executable.
3. Making `cpio` and `tar` symlinks makes it easier to replace them with `bsdcpio` and `bsdtar` respectively (think of all the CVEs avoided!).

### CODE REVIEW

**Review by thesamesam — COMMENTED** [review]
I haven't tested it but I'm fine with the concept.

File: app-arch/tar/tar-1.34-r1.ebuild [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+	dosym grmt.8 /usr/share/man/man8/rmt.8
+
+	mv "${ED}"/usr/sbin/backup{,-tar} || die
+	mv "${ED}"/usr/sbin/restore{,-tar} || die
+
+	if use minimal ; then
+		find "${ED}"/etc "${ED}"/*bin/ "${ED}"/usr/*bin/ \
+			-type f -a '!' '(' -name tar -o -name tar ')' \
+			-delete || die
+	fi
```
Comment (ionenwks) [inline app-arch/tar/tar-1.34-r1.ebuild:79]: Things get messy with USE=minimal
```
 * QA Notice: Symbolic link /usr/sbin/rmt points to /usr/sbin/grmt which does not exist.
 * QA Notice: Symbolic link /bin/tar points to /bin/gtar which does not exist.
```

File: app-arch/cpio/cpio-2.13-r3.ebuild [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+EAPI=7
+
+DESCRIPTION="A file archival tool which can also read and write tar files"
+HOMEPAGE="https://www.gnu.org/software/cpio/cpio.html"
+SRC_URI="mirror://gnu/cpio/${P}.tar.bz2"
+
+LICENSE="GPL-3+"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~loong ~m68k ~mips ~ppc ~ppc64 ~riscv ~s390 ~sparc ~x86 ~x64-cygwin ~amd64-linux ~x86-linux ~ppc-macos ~x64-macos ~sparc-solaris ~sparc64-solaris ~x64-solaris ~x86-solaris"
```
Comment (ionenwks) [inline app-arch/cpio/cpio-2.13-r3.ebuild:12]: Unsure for status on this, but note =cpio-2.13 is masked (without `~`) and a keyworded revbump exposes cpio-2.13.

**Review by ionenwks — COMMENTED** [review]
No strong feelings but I don't see a problem with this either, goes in-hand with gmake that we already do.

File: app-arch/cpio/cpio-2.13-r3.ebuild (hunk 2 above)
Comment (mgorny) [reply app-arch/cpio/cpio-2.13-r3.ebuild:12]: Ok, I'll just keep the unkeyworded version.

File: app-arch/tar/tar-1.34-r1.ebuild (hunk 1 above)
Comment (mgorny) [reply app-arch/tar/tar-1.34-r1.ebuild:79]: I wonder if anyone actually read this shitty code, given that it checks for `-name tar -o -name tar`. Anyway, gonna fix.

**Pull request closed without merging**